src/MentionsInput.js

```diff
@@ -280,6 +319,143 @@ class MentionsInput extends React.Component {
     }
   }
 
+  handlePaste(event) {
+    if (event.target !== this.inputRef) {
+      return
+    }
+
+    event.preventDefault()
+
+    const { selectionStart, selectionEnd } = this.state
+    const { value, children } = this.props
+
+    const config = readConfigFromChildren(children)
+
+    const realStartIndex = mapPlainTextIndex(
+      value,
+      config,
+      selectionStart,
+      'START'
+    )
+    const realEndIndex = mapPlainTextIndex(value, config, selectionEnd, 'END')
+
+    const pastedMentions = event.clipboardData.getData('text/react-mentions')
+    const pastedData = event.clipboardData.getData('text/plain')
+
+    const newValue = spliceString(
+      value,
+      realStartIndex,
+      realEndIndex,
+      pastedMentions || pastedData
+    )
+    const newPlainTextValue = getPlainText(newValue, config)
+
+    const eventMock = { target: { ...event.target, value: newValue } }
+
+    this.executeOnChange(
+      eventMock,
+      newValue,
+      newPlainTextValue,
+      getMentions(newValue, config)
+    )
+  }
+
+  handleCopy(event) {
+    if (event.target !== this.inputRef) {
+      return
+    }
+
+    const { selectionStart, selectionEnd } = this.state
+    const { children, value } = this.props
+
+    const config = readConfigFromChildren(children)
+
+    const realStartIndex = mapPlainTextIndex(
+      value,
+      config,
+      selectionStart,
+      'START'
+    )
+    const realEndIndex = mapPlainTextIndex(value, config, selectionEnd, 'END')
+
+    event.clipboardData.setData(
+      'text/plain',
+      event.target.value.slice(selectionStart, selectionEnd)
+    )
+    event.clipboardData.setData(
+      'text/react-mentions',
+      value.slice(realStartIndex, realEndIndex)
+    )
+
+    event.preventDefault()
+  }
+
+  handleCut(event) {
+    if (event.target !== this.inputRef) {
+      return
+    }
+
+    const { selectionStart, selectionEnd } = this.state
+    const { children, value } = this.props
+
+    const config = readConfigFromChildren(children)
+
+    const realStartIndex = mapPlainTextIndex(
+      value,
+      config,
+      selectionStart,
+      'START'
+    )
+    const realEndIndex = mapPlainTextIndex(value, config, selectionEnd, 'END')
+
+    event.clipboardData.setData(
+      'text/plain',
+      event.target.value.slice(selectionStart, selectionEnd)
+    )
+    event.clipboardData.setData(
+      'text/react-mentions',
+      value.slice(realStartIndex, realEndIndex)
+    )
+
+    event.preventDefault()
+
+    const mentions = getMentions(value, config)
+
+    const startMention = mentions.find(
+      mention =>
+        mention.plainTextIndex <= selectionStart &&
+        mention.plainTextIndex + mention.display.length > selectionStart
+    )
+    const endMention = mentions.find(
+      mention =>
+        mention.plainTextIndex <= selectionEnd &&
+        mention.plainTextIndex + mention.display.length > selectionEnd
+    )
+
+    const plainStartIndex = startMention
```

Comment:
Instead of the tedious calculation of `newPlainTextValue` could as well just use:

```
getPlainText(
  newValue,
  readConfigFromChildren(this.props.children)
)
```

Reply:
Thanks!